### DISCUSSION TITLE
Do we have to use powersync ?

### DISCUSSION THREAD

**Discussion by volkankoc** [itsezlife/flutter-instagram-offline-first-clone#140] [Q&A] [answered]
Do we have to use powersync

I want to make this application. But I don't quite understand why powersync is used.
How to do without powersync

**Comment by itsezlife** [comment]
Please, next time refer to a discussion section in this github repository with such non-techical questions not regarding any code issues. Thanks.

**Comment by itsezlife** [comment, marked as answer]
Well, the PowerSync in this project is a comprehensive extension to the Supabase to enable offline-first capabilities in the application.

You can skip the part with PowerSync and do exactly the same queries with Supabase, but in order to enable real-time with Supabase only, you need to accordingly enable real-time for specific tables in the database. Just Google it and it's done.

**Comment by volkankoc** [comment]
Understood thanks.
So this can be done without powersync.
Then you used powersync to avoid using realtime. Did I get right

**Reply by itsezlife** [reply]
No, I didn't use PowerSync to avoid using real-time.

The PowerSync ensures both real-time and offline-first in your application. It's built on top of the database you've chosen, in this application it is Supabase, and PowerSync internally uses an Async SQLite local database to handle offline-first and real-time. In the background it syncs all the data between local and remote database, you've linked it with.

PowerSync is a comprehensive tool to ensure both real-time and offline-first in your application. If you don't need offline-first, then use either only Supabase, or any other services, your own backend solutions, whatever you like the most.

This tutorial is completely intended to showcase the capabilities of the PowerSync+Supabase. If you don't want to use PowerSync, then just stick with what you like the most, but keep in mind that you'll have to re-engineer a bunch of things yourself. I hope you understand it.

**Comment by volkankoc** [comment]
I understand How do I do it without using Powersync? Can you help with this?
This is my main goal in this project. Fcm notifications and usage.
